pinot-core/src/main/java/org/apache/pinot/core/plan/FilterPlanNode.java

```diff
@@ -255,6 +255,9 @@ private BaseFilterOperator constructPhysicalOperator(FilterContext filter, int n
                   || textIndexReader instanceof NativeMutableTextIndex) {
                 throw new UnsupportedOperationException("TEXT_MATCH is not supported on native text index");
               }
+              if (textIndexReader == null) {
+                throw new UnsupportedOperationException("TEXT_MATCH is not supported when there is no text index");
```

Comment:
This moves the correctness check from execution to parsing time, which should be better, but makes `ExplainPlanQueriesTest` fail.

As indicated by its name, this tests does execute `explain plan` instead of the actual query. This is why the test didn't fail before. As this PR shouldn't be related to `text_match` but `regexp_like`, I'm going to remove this change for simplicity